specs/eventing/data-plane.md

```diff
@@ -1,125 +1,171 @@
+(Unavailable for Legal Reasons)? -->
+
+Recipients MUST be able to handle duplicate delivery of events and MUST accept
+delivery of duplicate events, as the event acknowledgement could have been lost
+in return to the sender. Event recipients MUST use the
+[`source` and `id` attributes](https://github.com/cloudevents/spec/blob/v1.0.1/spec.md#required-attributes)
+to determine duplicate events (see [observability](#observability) for an
+example case where other event attributes may vary from one delivery attempt to
+another).
+
+Where possible, event senders SHOULD re-attempt delivery of events where the
+HTTP request failed or returned a retryable status code. It is RECOMMENDED that
+event senders implement some form of congestion control (such as exponential
+backoff) when managing retry timing. This specification does not document any
+specific congestion control algorithm or parameters.
+[Brokers](./overview.md#broker) and [Channels](./overview.md#channel) MUST
+implement congestion control and MUST implement retries.
 
 ### Observability
 
-CloudEvents received by Sink MAY have
-[Distributed Tracing Extension Attribute](https://github.com/cloudevents/spec/blob/v1.0/extensions/distributed-tracing.md).
-
-### Event reply contract
-
-An event sender supporting event replies SHOULD include a `Prefer: reply` header
-in delivery requests to indicate to the sink that event reply is supported. An
-event sender MAY ignore an event reply in the delivery response if the
-`Prefer: reply` header was not included in the delivery request.
-
-An example is that a Broker supporting event reply sends events with an
-additional header `Prefer: reply` so that the sink connected to the Broker knows
-event replies will be accepted. While a source sends events without the header,
-in which case the sink may assume that any event reply will be dropped without
-error or retry attempt. If a sink wishes to ensure the reply events will be
-delivered, it can check for the existence of the `Prefer: reply` header in the
-delivery request and respond with an error code if the header is not present.
-
-### Data plane contract for Sources
-
-See [Source Delivery specification](sources.md#source-event-delivery)
-for details.
-
-### Data plane contract for Channels
-
-See [Channel Delivery specification](channel.md#data-plane) for details.
-
-### Data plane contract for Brokers
-
-See [Broker Delivery specification](broker.md)
-
-## Changelog
-
-- 2020-04-20: `0.13.x release`: initial version that documents common contract
-  for sinks, sources, channels and brokers.
+Event senders MAY add or update CloudEvents attributes before sending to
+implement observability features such as tracing; in particular, the
+[`traceparent` and `tracestate` distributed tracing attributes](https://github.com/cloudevents/spec/blob/v1.0/extensions/distributed-tracing.md)
+may be modified in this way for each delivery attempt of the same event.
```

Comment:
Do not link that spec, we're not following it. Link the W3C trace context spec, that's the one we accept and use broadly.

Reply:
Yeah that's a long standing debate in the cloudevents community, and we're not following it at all... The wording of the spec is unclear and it's even more complicated understand how to use it and how to keep the headers in sync. As you see in the cloudevents repo, I even proposed to remove that subspec.

Linking to the trace context should be fine IMO, because it reflects the status of our implementation and it's widely accepted and used as spec, compared to the cloudevent thingy.

Reply:
Yeah, I pointed out that was a mess a while ago; I can _also_ link directly to the w3c in case a future version of the CE spec removes the tracecontext content.

Reply:
but we don't support the cloudevents distributed tracing extension at all, if you send events with `ce-traceparent` and `ce-tracecontext`, the broker won't recognize it, so why link it in first instance?

Reply:
It looks like this is https://github.com/cloudevents/sdk-go/pull/634

Is it worth updating the CloudEvent spec to indicate that the extension is deprecated?

cc @duglin